provider/maas/environ.go

```diff
@@ -902,26 +902,31 @@ func (*maasEnviron) MaintainInstance(args environs.StartInstanceParams) error {
 func (environ *maasEnviron) StartInstance(args environs.StartInstanceParams) (
 	*environs.StartInstanceResult, error,
 ) {
-	var availabilityZone, nodeName, systemId string
+	availabilityZone := args.AvailabilityZone
+	var nodeName, systemId string
 	if args.Placement != "" {
 		placement, err := environ.parsePlacement(args.Placement)
 		if err != nil {
 			return nil, err
 		}
+		// NOTE(axw) we wipe out args.AvailabilityZone if the
+		// user specified a specific node or system ID via
+		// placement, as placement must always take precedence.
 		switch {
-		case placement.zoneName != "":
-			availabilityZone = placement.zoneName
```

Comment:
I'm a little concerned that this would break "--to zone=BLAH" but as long as we're testing that we're doing the right thing, I don't care whether this is here, or whether it triggers in DeriveAZ.

Reply:
I've just confirmed with a live test that we do the right thing (i.e. --to zone=azone fails).